### ISSUE TITLE
Wrap wsconnection#send in a try/catch to avoid exception in window between ws closing and 'close' event

### ISSUE THREAD

**Issue by SimonWoolf** [ably/ably-js#397]
eg
```
Ably: ConnectionManager.sendImpl(): Unexpected exception in transport.send(): Error: not opened
    at WebSocket.send (/home/ubuntu/node_modules/ably/node_modules/ws/lib/WebSocket.js:219:16)
    at WebSocketTransport.send (/home/ubuntu/node_modules/ably/common/lib/transport/websockettransport.js:89:16)
    at Protocol.send (/home/ubuntu/node_modules/ably/common/lib/transport/protocol.js:44:18)
    at ConnectionManager.sendImpl (/home/ubuntu/node_modules/ably/common/lib/transport/connectionmanager.js:1301:24)
    at ConnectionManager.send (/home/ubuntu/node_modules/ably/common/lib/transport/connectionmanager.js:1277:9)
    at RealtimeChannel.sendMessage (/home/ubuntu/node_modules/ably/common/lib/client/realtimechannel.js:281:26)
    at RealtimeChannel._publish (/home/ubuntu/node_modules/ably/common/lib/client/realtimechannel.js:93:10)
    at /home/ubuntu/node_modules/ably/common/lib/client/realtimechannel.js:77:9
    at /home/ubuntu/node_modules/ably/common/lib/types/message.js:129:6
    at Function.Message.encode (/home/ubuntu/node_modules/ably/common/lib/types/message.js:115:4)
```

**Comment by SimonWoolf** [comment]
Relatedly, seen on many connections simultaneously during a loadtest:
```
/home/ubuntu/node_modules/ws/lib/WebSocket.js:219
    else throw new Error('not opened');
         ^

Error: not opened
    at WebSocket.send (/home/ubuntu/node_modules/ws/lib/WebSocket.js:219:16)
    at WebSocketTransport.send (/home/ubuntu/node_modules/ably/common/lib/transport/websockettransport.js:88:16)
    at WebSocketTransport.Transport.requestDisconnect (/home/ubuntu/node_modules/ably/common/lib/transport/transport.js:176:8)
    at WebSocketTransport.Transport.disconnect (/home/ubuntu/node_modules/ably/common/lib/transport/transport.js:48:9)
    at WebSocketTransport.Transport.onIdleTimerExpire (/home/ubuntu/node_modules/ably/common/lib/transport/transport.js:206:9)
    at Timeout._onTimeout (/home/ubuntu/node_modules/ably/common/lib/transport/transport.js:194:10)
    at ontimeout (timers.js:380:14)
    at tryOnTimeout (timers.js:244:5)
    at Timer.listOnTimeout (timers.js:214:5)
```
That's the idleTimer expiring (so if the lib goes 25s without seeing a heartbeat), requesting a disconnect. This suggests that this isn't a race condition, but rather the websocket has been closed but without the onclose or onerror handlers having been called (which would have disposed of the transport).

**Comment by SimonWoolf** [comment]
Fixed by ce89be8
